### PR TITLE
Add a space between Mantid and Workbench/Plot on Win

### LINKED ISSUE #27484: Consider Renaming MantidWorkbench -> Mantid Workbench
Currently on Windows typing `Workbench`or any variant will not return any results related to Mantid. 
So most users will just type `mantid` hit enter and get Mantid Plot (which is what we want)

We should investigate:
- [x] Can we change MantidWorkbench -> Mantid Workbench without causing breakage
- [x] We **must** ensure that typing Mantid and hitting enter still brings up plot (currently)
- [x] Check that putting the space means that Workbench works correctly

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Adds a space between the word Mantid and Workbench or Plot on Windows.

This allows Windows search to find results based on the word plot or Workbench, whereas before Mantid was the only word it selected on.

I'm keen to get this in really early in the release window, so we have a long time for it to stew since it involves the primary way users access our software.

Some notes from my research following on in #27484 :
- This only affects Windows (because of how CMake is wired up) but other platforms are not affected at all. This includes the application names, only shortcuts names (i.e. start menu) is changed.
- After this change Best Match seems to 'work' correctly. For me it changed to Workbench when I typed 'Mantid' in. After manually selecting plot 3/4 times typing 'Mantid' would default open Plot instead.
- We have no ability to hint or control the order Window's start menu will order in, so it's purely based on whatever secret heuristic Microsoft use to order apps in
- Users will now be able to type 'Plot' or 'Workbench' if they have strong feelings about one or another. This could help adoption, as typing 'Mantid' gets you Plot by default, and if you really feel strongly against workbench you can just type plot to know (or let Windows figure it out).

**To test:**
- You must have CPack configured, see the dev instructions on building Windows Installers 
- Run CMake with -DENABLE_CPACK=ON -DCPACK_PACKAGE_SUFFIX=""
- (Or alternatively in the CMake GUI set ENABLE_CPACK and change CPACK_PACKAGE_SUFFIX to nothing)
- Build a release build, a debug build is not supported for packaging
- Uninstall Mantid release whilst you wait
- Build a Windows Installer using the installer instructions
- Install your bespoke version and test out the start menu
- Ensure you uninstall and delete your bespoke version, so you don't get it confused with an official release.
- Try this for a nightly or unstable release and check `plot`, `workbench` and `mantid` still work correctly

Fixes #27484 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DanNixon** [comment]
Do the same issues also apply to other platforms (GNOME, whatever OS X has, etc.)?

**Comment by martyngigg** [comment]
@DanielMurphy22 Could you check macOS?

**Comment by DavidFair** [comment]
So asking around the room for Ubuntu:

KDE - Does the right thing
Gnome - Does not, so it will need the same fix as Windows

**Comment by martyngigg** [comment]
Why is searching in a string so hard??

We don't produce anything separate for KDE/Gnome so we'll just have to do it for all Linux.

**Comment by DavidFair** [comment]
I'll mark this as not ready and add a fix for Ubuntu, and pending Daniel's comments OS X

**Comment by DanielMurphy22** [comment]
It's fine on MacOS!

**Comment by martyngigg** [comment]
Should we put the space in the title anyway for consistency? The changes go in to the `plist` for Mac: https://github.com/mantidproject/mantid/blob/6577e594e32f54f35fcf60cf2301db9508a821d3/qt/applications/workbench/CMakeLists.txt#L110

**Comment by DanNixon** [comment]
> Why is searching in a string so hard??

DE developers probably just pick something easy to implement or fast because typically don't care about scenarios like this.

**Comment by martyngigg** [comment]
I can confirm this works well on Windows.

**Comment by DavidFair** [comment]
@DanielMurphy22 - could you give the OSX changes a look and install, I'm not 100% with packaging for the platform so it would be good to check if Plot/Notebook/Workbench all have a space now

@martyngigg I can't see the relevant CMake config for Linux, maybe I'm missing something?

**Comment by martyngigg** [comment]
@DavidFair https://github.com/mantidproject/mantid/blob/48284f2073453f3844e6159fad8048cc2aee3e61/CMakeLists.txt#L301 is the section for Linux

File: CMakeLists.txt [hunk 1]
```diff
@@ -333,7 +333,7 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
       set(_icon_filename ${_icon_filename}${_icon_suffix})
     endif()
     install_desktop_files(mantidplot${CPACK_PACKAGE_SUFFIX}.desktop
-                          MantidPlot${_app_name_suffix}
+                          Mantid Plot${_app_name_suffix}
```
Comment (DanNixon) [inline CMakeLists.txt]: Do we also want a space between `Plot` and `${_app_name_suffix}`?

Comment (DavidFair) [reply CMakeLists.txt]: 🤦‍♂ I knew I'd forgotten something - we have this on Windows already

**Comment by DanielMurphy22** [comment]
> @DanielMurphy22 - could you give the OSX changes a look and install, I'm not 100% with packaging for the platform so it would be good to check if Plot/Notebook/Workbench all have a space now

I've tried this on MacOS and, although Workbench didn't seem to package correctly, neither Plot or WB had spaces anywhere I looked, or when trying to search.

**Comment by DavidFair** [comment]
So we've tested a couple of iterations of the CMake configuration on OSX to no avail. It was also pointed out that we don't create separate installs for nightly and unstable builds either.

I've opened an issue #27524 to fix both. However as `Workbench` works correctly on OSX I don't want to hold up this PR for long, since I would rather know about any subtle breakages it causes early into the release cycle.
For that reason I'm going to revert the OSX changes and just leave Windows and Linux

**Comment by martyngigg** [comment]
@DavidFair Having separate installs for macOS has been on the plan for a while but never high enough priority. It might be beneficial with the Python 3 switch though.

**Review by martyngigg — APPROVED** [review]
Confirmed that this works well on Linux.

**Pull request merged**